services/queue/src/utils.js

```diff
@@ -42,86 +41,108 @@ const artifactUtils = {
   async expire({ db, publicBucket, privateBucket, ignoreError, monitor, expires }) {
     let count = 0;
 
-    // Get all expired artifacts using a handler function, operating on one
-    // page of results at a time.  This does not use paginatedIterator because
-    // that would result in a much slower one-by-one expiration, instead of
-    // the parallel handling of each batch.
-    const getExpiredArtifacts = async (expires, handler) => {
-      let after_task_id_in = null;
```

Comment:
we no longer need to mark the "last read" position, since new `get_expired_artifacts_for_deletion` always returns _any_ _first_ records that db gives us that satisfies `expires < ts` filter. 
Those returned rows can even be random, but we don't care since we know we are going to remove them afterwards anyway and they will not appear in any query again